src/descformats/base.py

```diff
@@ -0,0 +1,284 @@
+    """DataHandle for a table written to fits"""
+    suffix = 'fits'
+    format = "http://edamontology.org/format_2333"    
+    required_columns = []
+
+    @classmethod
+    def _open(cls, path, mode, **kwargs):
+        import fitsio   # pylint: disable=import-outside-toplevel
+        # Fitsio doesn't have pure 'w' modes, just 'rw'.
+        # Maybe we should check if the file already exists here?
+        if mode == 'w':
+            mode = 'rw'
+        return fitsio.FITS(path, mode=mode, **kwargs)
+
+    def missing_columns(self, columns, hdu=1):
+        """
+        Check that all supplied columns exist
+        and are in the chosen HDU
+        """
+        if self.data is not None:
+            ext = self.data
+            found_cols = list(ext.keys())
+        elif self.fileObj is None:
+            self.open(mode='r')
+            ext = self.fileObj[hdu]
+            found_cols = ext.get_colnames()
+        missing_columns = [col for col in columns if col not in found_cols]
+        return missing_columns
+
+    def validate(self):
+        """Check that the catalog has all the required columns and complain otherwise"""
+        # Find any columns that do not exist in the file
+
+        missing = self.missing_columns(self.required_columns)
+
+        # If there are any, raise an exception that lists them explicitly
+        if missing:
+            text = "\n".join(missing)
+            raise FileValidationError(f"These columns are missing from FITS file {self.path}:\n{text}")
+
+
+class PqHandle(TableHandle):
+    """DataHandle for a parquet table"""
+    suffix = 'pq'
+
+    @classmethod
+    def _close(cls, fileObj, **kwargs):
+        pass
+
+
+class QPHandle(DataHandle):
+    """DataHandle for qp ensembles
+    """
+    suffix = 'hdf5'
+
+    @classmethod
+    def _open(cls, path, mode, **kwargs):
+        """Open and return the associated file
+        Notes
+        -----
+        This will simply open the file and return a file-like object to the caller.
+        It will not read or cache the data
+        """
+        return tables_io.io.io_open(path, mode=mode, **kwargs)  #pylint: disable=no-member
+
+    @classmethod
+    def _read(cls, path, **kwargs):
+        """Read and return the data from the associated file """
+        return qp.read(path)
+
+    @classmethod
+    def _write(cls, data, path, **kwargs):
+        """Write the data to the associatied file """
+        return data.write_to(path)
+
+    @classmethod
+    def _initialize_write(cls, data, path, data_lenght, **kwargs):
+        comm = kwargs.get('communicator', None)
+        return data.initializeHdf5Write(path, data_lenght, comm)
+
+    @classmethod
+    def _write_chunk(cls, data, fileObj, groups, start, end, **kwargs):
+        return data.writeHdf5Chunk(fileObj, start, end)
+
+    @classmethod
+    def _finalize_write(cls, data, fileObj, **kwargs):
+        return data.finalizeHdf5Write(fileObj)
+
+
+HDFFile = Hdf5Handle
```

Comment:
We don't use HDF5 files as pure table files in TXPipe.  There's various other cases like maps and things like that.

I think we might want to split out HDFTable and HDFFile as two separate things?

Reply:
Yes, good point, that makes sense.